common/notification/messages.ts

```diff
@@ -42,6 +48,68 @@ export async function getMessageForNotification(
     return { body, headline, modalText, navigateTo: translation.navigateTo, type: notification.type as NotificationType };
 }
 
+type CachedMessageTemplate = {
+    type: NotificationType;
+    body: (context: NotificationContext) => string;
+    headline: (context: NotificationContext) => string;
+    navigateTo?: (context: NotificationContext) => string;
+    modalText?: (context: NotificationContext) => string;
+};
+
+const cachedTemplates = new Map</* notification id + language */ string, CachedMessageTemplate | null /* = no message */>();
```

Comment:
This would mean that we have to restart the backend for every notification change.
This doesn't happen very often, but it somewhat works against the initial idea to have a dynamic system that can be configured via retool.
Theoretically, we could create a TTL cache to automatically refresh the cache from time to time.

Reply:
Not quite, when a Message Template is updated with setMessageTranslation, we invalidate the cache ... So no need to restart. There is also an integration test for that case :)

Reply:
For such systems, I'm mostly afraid about the future. We must remember to invalidate the cache every time we update the translation.
In this case, it will not be a problem, but as soon as updates are done in other files, it's getting complex :)